src/overlays/actors/ovl_Bg_Jya_Bigmirror/z_bg_jya_bigmirror.c

```diff
@@ -26,21 +28,242 @@ const ActorInit Bg_Jya_Bigmirror_InitVars = {
+                curCobraInfo->rotY = curCobraInfo->cobra->actor.shape.rot.y;
+
+                if (curCobraInfo->rotY == curSpawnData->solvedRotY) {
+                    this->puzzleState |= cobraPuzzleFlags[i];
+                } else {
+                    this->puzzleState &= ~cobraPuzzleFlags[i];
+                }
+
+                if (curCobraInfo->cobra->actor.update == 0) {
+                    // Cobra deleted
+                    osSyncPrintf("Error : コブラ削除された (%s %d)\n", "../z_bg_jya_bigmirror.c", 203);
+                }
+            } else {
+                curCobraInfo->cobra = (BgJyaCobra*)Actor_SpawnAsChild(
+                    &globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_BG_JYA_COBRA, curSpawnData->pos.x,
+                    curSpawnData->pos.y, curSpawnData->pos.z, 0, curCobraInfo->rotY, 0, curSpawnData->params);
+                this->actor.child = NULL;
+
+                if (&curCobraInfo->cobra->actor == NULL) {
+                    // Cobra generation failed
+                    osSyncPrintf("Error : コブラ発生失敗 (%s %d)\n", "../z_bg_jya_bigmirror.c", 221);
+                }
+            }
+        }
+    } else { // In statue room
+
+        for (i = 0; i < 2; i++) {
+            curCobraInfo = &this->cobraInfo[i];
+            if (curCobraInfo->cobra != NULL) {
+                if (curCobraInfo->cobra->actor.child != NULL) {
+                    Actor_Kill(curCobraInfo->cobra->actor.child);
+                    curCobraInfo->cobra->actor.child = NULL;
+                }
+                Actor_Kill(&curCobraInfo->cobra->actor);
+                curCobraInfo->cobra = NULL;
+            }
+        }
+    }
+}
+
+void BgJyaBigmirror_SetBombiwaFlag(Actor* thisx, GlobalContext* globalCtx) {
+    BgJyaBigmirror* this = THIS;
+
+    if (Flags_GetSwitch(globalCtx, 0x29) != 0) {
+        this->puzzleState |= 4;
+    } else {
+        this->puzzleState &= ~4;
+    }
+}
+
+void BgJyaBigmirror_HandleMirRay(Actor* thisx, GlobalContext* globalCtx) {
+    static s16 sMirRayParamss[] = { 0x0005, 0x0007, 0x0008 };
+    static Vec3f sMirRayPoss[] = {
+        { 60, 1802, -1102 },
+        { -560, 1800, -310 },
+        { 60, 1800, -310 },
+    };
+    BgJyaBigmirror* this = THIS;
+
+    s32 inTopRooms;
+    s32 lightBeamToggles[3];
+    s32 puzzleSolved;
+    s32 objBankIndex;
+    s32 i;
+
+    objBankIndex = Object_GetIndex(&globalCtx->objectCtx, OBJECT_MIR_RAY);
+
+    if ((objBankIndex < 0) || (objBankIndex != this->mirRayObjIndex)) {
+        this->lightBeams[2] = NULL;
+        this->lightBeams[1] = NULL;
+        this->lightBeams[0] = NULL;
+    } else {
+        puzzleSolved = (this->puzzleState & 0x18) != 0; // In second room or puzzle progress made
+
+        if (puzzleSolved != 0) {
+            puzzleSolved = (this->puzzleState & 2) != 0; // Second cobra solved
+
+            if (puzzleSolved != 0) {
+                puzzleSolved = (this->puzzleState & 1) != 0; // First cobra solved
+            }
```

Comment:
should be able to omit !=0 on all of these

Reply:
I know this looks bizarre, but it actually does `sltu    v0,zero,v0` for each one, so they have to be there. Adding a comment

Reply:
silly thing to try, !! on the outside

Reply:
Okay, that worked after doing chained assignment, getting rid of a temp and permuting the stack variables a bit